src/main/java/org/wildfly/security/auth/realm/ldap/X509EvidenceVerifier.java

```diff
@@ -124,7 +125,11 @@ public boolean verifyCertificate(X509Certificate certificate, Attributes attribu
             Attribute attribute = attributes.get(ldapAttribute);
             final int size = attribute.size();
             for (int i = 0; i < size; i++) {
-                if (certificate.getSubjectDN().getName().equals(attribute.get(i))) {
+
+                X500Principal certSubjectX500Principal = certificate.getSubjectX500Principal();
+                X500Principal attSubjectX500Principal = new X500Principal((String)attribute.get(i));
```

Comment:
@darranl In think we will never get null retrieving them from LDAP, because if the attribute x509subject is in the LDAP server and it doesn't have a value, the resulting value will be an empty String (by RFC statement https://tools.ietf.org/html/draft-klasen-ldap-x509certificate-schema-03#section-4.1.6). We can add this to ensure robustness against invalid configurations but then I think we should do the same with the others attributes retrieved in the same way in the same class. Agree?